### DISCUSSION TITLE
[New models] IPAdapter composition [SD1.5/SDXL]

### DISCUSSION THREAD

**Discussion by huchenlei** [Mikubill/sd-webui-controlnet#2781] [General]
Original project: https://huggingface.co/ostris/ip-composition-adapter

> This adapter for Stable Diffusion 1.5 and SDXL is designed to inject the general composition of an image into the model while mostly ignoring the style and content. Meaning a portrait of a person waving their left hand will result in an image of a completely different person waving with their left hand.

### Usage
Just use it as any other ip-adapter models.

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/20929282/cb7023a2-b356-4841-be96-c82fa48ce871)
SDXL Output:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/20929282/33dd40d4-6e01-4a4e-94e1-b36a6b52754b)
SDXL infotext:
```
1girl, absurdres, highres, (masterpiece:1.2), (best quality, highest quality),
Negative prompt: (lowres, low quality, worst quality:1.2), (text:1.2), watermark, (frame:1.2), deformed, ugly, deformed eyes, blur, out of focus, blurry, deformed cat, deformed, photo, anthropomorphic cat, monochrome, photo, pet collar, gun, weapon, blue, 3d, drones, drone, buildings in background, green
Steps: 20, Sampler: DPM++ 2M, Schedule type: Karras, CFG scale: 7, Seed: 1220524552, Size: 1000x1000, Model hash: 5d973c1f8a, Model: animagine-xl-2.0, Clip skip: 2, ControlNet 0: "Module: ip-adapter-auto, Model: ip-adapter_plus_composition_sdxl [7af1e3cf], Weight: 1, Resize Mode: Just Resize, Low Vram: False, Processor Res: 512, Threshold A: 0.5, Threshold B: 0.5, Guidance Start: 0, Guidance End: 1, Pixel Perfect: False, Control Mode: Balanced, Hr Option: Both, Save Detected Map: True", Version: v1.9.0-46-gd1998d74
```

SD1.5 Output
![00019-1220524552](https://github.com/Mikubill/sd-webui-controlnet/assets/20929282/2adf7bc0-25eb-4861-b768-b562f8106d2f)
```
1girl, absurdres, highres, (masterpiece:1.2), (best quality, highest quality),
Negative prompt: (worst quality:2), (low quality:2), (normal quality:2), lowres, normal quality, ((monochrome)), ((grayscale)), skin spots, acnes, skin blemishes, age spot, backlight,(ugly:1.331), (duplicate:1.331), (morbid:1.21), (mutilated:1.21), (tranny:1.331), mutated hands, (poorly drawn hands:1.331), blurry, (bad anatomy:1.21), (bad proportions:1.331), extra limbs, (more than 2 nipples:1.331), (missing arms:1.331), (extra legs:1.331), (fused fingers:1.61051), (too many fingers:1.61051), (unclear eyes:1.331), bad hands, missing fingers, extra digit, (futa:1.1), bad body, pubic hair, glans, easynegative,more than 2 tits, ng_deepnegative_v1_75t,(big fee:1),more than 2 feet,incorrect feet, easynegative,
Steps: 20, Sampler: DPM++ 2M, Schedule type: Karras, CFG scale: 7, Seed: 1220524552, Size: 512x512, Model hash: b42b09ff12, Model: cetusMix_v4, Clip skip: 2, ControlNet 0: "Module: ip-adapter_clip_h, Model: ip-adapter_plus_composition_sd15 [f910d03b], Weight: 1, Resize Mode: Just Resize, Low Vram: False, Processor Res: 512, Threshold A: 0.5, Threshold B: 0.5, Guidance Start: 0, Guidance End: 1, Pixel Perfect: False, Control Mode: Balanced, Hr Option: Both, Save Detected Map: True", TI hashes: "easynegative: c74b4e810b03, ng_deepnegative_v1_75t: 54e7e4826d53", Version: v1.9.0-46-gd1998d74
```

### Model download
https://huggingface.co/ostris/ip-composition-adapter/tree/main
You need to rename model files to `ip-adapter_plus_composition_sd15.safetensors` and `ip-adapter_plus_composition_sdxl.safetensors` for sd-webui-controlnet extension to properly detect them.

**Comment by CCpt5** [comment]
> ### Model download
> https://huggingface.co/ostris/ip-composition-adapter/tree/main You need to rename model files to `ip-adapter_plus_composition_sd15.safetensors` and `ip-adapter_plus_composition_sdxl.safetensors` for sd-webui-controlnet extension to properly detect them.

@huchenlei 
I had an issue w/ the sd1.5 model not showing up even though I had renamed the files per your note (copy/pasted). Please check, but I believe the filename for SD1.5 actually has to be: ip-adapter_plus_composition_sd1.5.safetensors   - In your post above(quoted) you don't include a period in the sd 1.5 filename (Re: sd15.safetensors instead of sd1.5.safetensors) - The latter worked for me, the former as listed above failed to have the model show up. 

Thank you so much for this btw. One I got the model recognized it worked brilliantly!
